catalog/spatial/csw/spatial-csw-transformer/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/transformer/GmdTransformer.java

```diff
@@ -210,187 +306,596 @@ private void copyArgumentsToContext(MarshallingContext context,
+                        Double lon = Double.parseDouble(coordinateStrings[1]);
+                        Coordinate coordinate = new Coordinate(lat, lon);
+                        coordinates.add(coordinate);
+                    }
+                }
 
-        List<String> keywords = pathValueTracker.getAllValues(toPath(GmdMetacardType.KEYWORD_PATH));
-        List<String> topics =
-                pathValueTracker.getAllValues(toPath(GmdMetacardType.TOPIC_CATEGORY_PATH));
+                if (CollectionUtils.isNotEmpty(coordinates)) {
+                    // Close the polygon
+                    coordinates.add(coordinates.get(0));
+                    LinearRing linearRing =
+                            factory.createLinearRing(coordinates.toArray(new Coordinate[coordinates.size()]));
+                    String wkt = WKT_WRITER_THREAD_LOCAL.get()
+                            .write(factory.createPolygon(linearRing, null));
+                    if (wkt != null) {
+                        metacard.setAttribute(new AttributeImpl(Core.LOCATION, wkt));
+                    }
+                }
+            }
 
-        List<String> subjects = new ArrayList<>();
-        if (CollectionUtils.isNotEmpty(keywords)) {
-            subjects.addAll(keywords);
+        } catch (ParserConfigurationException | IOException | SAXException | XPathExpressionException e) {
+            LOGGER.debug(
+                    "Unable to parse location in XML document.  Metacard location will not be set.",
+                    e);
+            return;
         }
-        if (CollectionUtils.isNotEmpty(topics)) {
-            subjects.addAll(topics);
+    }
+
+    private void setMetacardDates(Metacard metacard,
+            final XstreamPathValueTracker pathValueTracker) {
+        try (InputStream inputStream = new ByteArrayInputStream(byteArray)) {
+            DocumentBuilderFactory domFactory = DocumentBuilderFactory.newInstance();
+            domFactory.setNamespaceAware(false);
+            DocumentBuilder builder = domFactory.newDocumentBuilder();
+            Document document = builder.parse(inputStream);
+
+            XPath xPath = XPathFactory.newInstance()
+                    .newXPath();
+            XPathExpression dataExpression = xPath.compile(GmdConstants.RESOURCE_DATE_PATH);
+            NodeList dataList = (NodeList) dataExpression.evaluate(document,
+                    XPathConstants.NODESET);
+
+            if (dataList != null && dataList.getLength() > 0) {
+                List<String> dateList = new ArrayList<>();
+                List<String> dateTypes =
+                        pathValueTracker.getAllValues(toPath(GmdConstants.CITATION_DATE_TYPE_PATH));
+
+                for (int i = 0; i < dataList.getLength(); i++) {
+                    Node node = dataList.item(i);
+                    String datestring = node.getTextContent()
+                            .trim()
+                            .replaceAll("(\\s|\\t)+", " ")
```

Comment:
\s already includes \t and \n: From java: \s => A whitespace character: [ \t\n\x0B\f\r]